### DISCUSSION TITLE
Does Docusaurus have content sections like Hugo?

### DISCUSSION THREAD

**Discussion by josmperez** [facebook/docusaurus#9142] [Q&A]
I am migrating a site from Hugo to Docusaurus and can't find any feature like [Hugo sections](https://gohugo.io/content-management/sections/).

This is an auto-generated tree that lists/links to all the pages in a directory.

Is this feature available now or on the roadmap?

**Comment by slorber** [comment]
In Docusaurus you can just put Markdown files in the `./docs` folder (including in nested subfolders) and they become available to an URL.

If you want files to appear in a sidebar tree, you need an autogenerated sidebar: https://docusaurus.io/docs/sidebar/autogenerated
This creates categories (our equivalent of Hugo sections?) for each `./docs` subfolder.

---

The initial/default Docusaurus template uses this, so I'm not sure what you don't understand exactly 🤷‍♂️

Yes we support this, it's pretty intuitive and explained in our 5min tutorial: https://tutorial.docusaurus.io/docs/tutorial-basics/create-a-document

Have you followed the 5min tutorial? This is the first thing you should do, try it with https://docusaurus.new without installing anything.

That's basically the first thing you learn: you can just add a file to your filesystem and it is automatically handled by Docusaurus.

**Reply by josmperez** [reply]
I understand how documents with auto-generated sidebars are created. Hugo sections aren't sidebars, they're shortcodes that can be inserted into a markdown page that insert a tree of all the child folders. They are useful for adding to index.md files so that the task of inserting a list of links into the page itself is automated, kept up-to-date. 

Since I haven't heard a reply, I think this is a situation where Hugo is probably ahead of Docusaurus at the moment. But I have already manually created all the index pages I need, so it's not a deal breakers.

**Reply by slorber** [reply]
I'm not sure understand what a Hugo section is, but we have a way to embed inside a page a tree/index of categories/documents: https://docusaurus.io/docs/sidebar/items#embedding-generated-index-in-doc-page

<img width="999" alt="CleanShot 2023-07-20 at 11 05 39@2x" src="https://github.com/facebook/docusaurus/assets/749374/078728a3-56aa-4af1-97e5-c1d6c7ac0da6">


Maybe it doesn't display the way you want (here we display only the 1st level subitems of the current category as cards) but you can build your own UI with React to display the tree data if you want, the data is accessible with these 2 React hooks:

```ts
// Get the whole sidebar tree you are inside of (from the root)
import {useDocsSidebar} from '@docusaurus/theme-common/internal';

// Get the sidebar subtree you are inside of
import {useCurrentSidebarCategory} from '@docusaurus/theme-common';
```

Note: even if you don't want to display a sidebar on your docs, the tree that we expose to you is still the same as the tree of the sidebar, it's the same data structure behind.
